libraries/AP_OSD/AP_OSD_Screen.cpp

```diff
@@ -1382,7 +1384,8 @@ void AP_OSD_Screen::draw_waypoint(uint8_t x, uint8_t y)
 
 void AP_OSD_Screen::draw_xtrack_error(uint8_t x, uint8_t y)
 {
-    backend->write(x, y, false, "%c%4d", SYM_XERR, (int)osd->nav_info.wp_xtrack_error);
+    backend->write(x, y, false, "%c", SYM_XERR);
+    draw_distance(x+1, y, (int)osd->nav_info.wp_xtrack_error);
```

Comment:
why cast to int when draw_distance takes float?

Reply:
I will fix..it was just me copying the old directly drawn object which was cast like that...will change it

Reply:
fixed